models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -215,7 +215,7 @@ public static function process(
         if ($deferred) {
             // only add the config to the TmpStore if necessary (e.g. if the config is auto-generated)
             if (!Config::exists($config->getName())) {
-                $pathInfo = trim($asset->getRealPath(), '/').'/'.$asset->getId() . '/';
+                $pathInfo = trim($asset->getRealPath(), '/').'/'.$asset->getId() . '/' . $config->getName();
```

Comment:
Unfortunately this doesn't cover different image formats (when using auto, e.g. WebP, AVIF, JPG) as well as different resolutions (`@2x`). 
That's why we used to utilize the `$storagePath` here, which includes both variables.

Reply:
I can confirm that. We are heavily relying on auto thumbs for responsive images and with #15198 they are really broken.
Guess creating the deferred config id using only the basepath and asset id is not enough?

Simplest example is using 
```
{{ pimcore_image("myImage", {
    "thumbnail": {
        "width": 200,
        "height": 200
    }
}) }}
{{ pimcore_image("myImage2", {
    "thumbnail": {
        "width": 300,
        "height": 400
    }
}) }}
```
Then try to open an image from the sources only by path - eg. currently displayed is webp -> use jpg.

Reply:
Filename is now added to deferred config id, and root-level assets are fixed too. 
Can you please double check.

Reply:
root-level assets work -> lgtm

I see that there are more TmpStore entries for the thumb configs -> lgtm

I have the same asset twice on the page with different aspect-ratios and methods (cover/frame) -> lgtm
Selecting an other image from the src-set seems to work -> lgtm


But it looks like the thumbnails are only created/served correctly at the second request.

Found this in dev.log. Hope this can help.
```
1. Request:

[2024-01-17T10:26:38.784925+01:00] request.INFO: Matched route "_pimcore_service_thumbnail". {"route":"_pimcore_service_thumbnail","route_parameters":{"_route":"_pimcore_service_thumbnail","_controller":"Pimcore\\Bundle\\CoreBundle\\Controller\\PublicServicesController::thumbnailAction","prefix":"18/","type":"image","assetId":"18","thumbnailName":"_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca","filename":"4-542-928-0001-DYN-Katalog.b6085414.webp"},"request_uri":"http://xxxxx.test/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp","method":"GET"} []

[2024-01-17T10:26:38.785701+01:00] app.DEBUG: Resolved pimcore context for path /18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp to default {"path":"/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp","context":"default"} []

[2024-01-17T10:26:38.808707+01:00] pimcore.ERROR: Unable to generate image thumbnail, see logs for details. [] []

=====
2. Request

[2024-01-17T10:36:04.645148+01:00] request.INFO: Matched route "_pimcore_service_thumbnail". {"route":"_pimcore_service_thumbnail","route_parameters":{"_route":"_pimcore_service_thumbnail","_controller":"Pimcore\\Bundle\\CoreBundle\\Controller\\PublicServicesController::thumbnailAction","prefix":"18/","type":"image","assetId":"18","thumbnailName":"_auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca","filename":"4-542-928-0001-DYN-Katalog.b6085414.webp"},"request_uri":"http://xxxxx.test/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp","method":"GET"} []

[2024-01-17T10:36:04.645320+01:00] app.DEBUG: Resolved pimcore context for path /18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp to default {"path":"/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp","context":"default"} []

[2024-01-17T10:36:04.650961+01:00] init.DEBUG: Resolved editmode to false {"editmode":"false","params":{"param":false,"adminRequest":false,"user":false}} []

[2024-01-17T10:36:04.654153+01:00] php.DEBUG: Warning: filemtime(): stat failed for /xxxxx/public/var/tmp/thumbnails/18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp {"exception":{"Symfony\\Component\\ErrorHandler\\Exception\\SilencedErrorContext":{"severity":2,"file":"/xxxxx/vendor/league/flysystem-local/LocalFilesystemAdapter.php","line":427,"trace":[{"file":"/xxxxx/vendor/league/flysystem/src/Filesystem.php","line":161,"function":"lastModified","class":"League\\Flysystem\\Local\\LocalFilesystemAdapter","type":"->"}],"count":1}}} []

[2024-01-17T10:36:04.817648+01:00] pimcore.DEBUG: Thumbnail /18/image-thumb__18___auto_98b4dc6ca8f2bb9c7061e0817cd3a2ca/4-542-928-0001-DYN-Katalog.b6085414.webp generated in 0.162113904953 seconds [] []
```

Reply:
Deleted all thumbnail files and open page:
![SCR-20240117-kpgx](https://github.com/pimcore/pimcore/assets/3617107/38bf15df-5fb7-4b9e-91ae-673c30e6befd)

Reply:
Hmm strange. Works for me with tje first request too. Could you try to debug on your end?

Reply:
hmm, I cannot reproduce that on my end. For me, the images work on the first load.
I would merge that now. If you have any additional findings, please feel free to create an issue, or even better already a PR to fix it ;-)

Reply:
Just one question:
The expiration date for the TmpStore entries should be higher (or even off ) ?
With full_page_cache enabled it can / will happen that the entry is removed and then the auto thumb cannot be created anymore.

Reply:
> Just one question: The expiration date for the TmpStore entries should be higher (or even off ) ? With full_page_cache enabled it can / will happen that the entry is removed and then the auto thumb cannot be created anymore.

it gets worse when using frontend_prefixes (cdn) because the frontend_prefix is only added if the thumbnail is already created. would be nice if the fullpage cache only kicks in when all thumbnails are generated for that page. it would also be nice if we would  have a command that generates thumbnails for alle thumbs in the tmpstore...

Reply:
@hadl I wouldn't keep the TmpStore items forever, but yes we can think about extending the time or even make it configurable. Could you create a PR for that? 

@bitbirddev I see the challenge that then a page will never get into the cache, because there can be so many different variants of a thumbnail, different formats, different resolutions, media queries, ... don't think that this would make sense in practice. 😬

Reply:
@brusch please take a look at #16467 for a detailed description of the problem